tools/pulumi-hcl-lint/lint_dangling_refs.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"github.com/pulumi/terraform/pkg/configs"
+)
+
+func lintDanglingRefs(ctx context.Context, mod *configs.Module, sink chan<- issue) {
+	vis := newDanglingRefVisitor()
+	for _, mr := range mod.ManagedResources {
+		vis.visitManagedResource(mr)
+	}
+	for _, r := range vis.dangling() {
+		if r.Name() == "value" && r.Token() == "each" {
```

Comment:
This is for loop variables, right? They are assumed to be defined?

Reply:
This came up as a false positive dangling ref detected in the one test example I was using.